File: tasks/implementations/go/tasks/ai_service_integration.go
package main

import (
    "fmt"
    "os"
    "path/filepath"
    "regexp"
    "strconv"
)

func RunAiServiceIntegration(size string, fixturesRoot string) {
    data, _ := os.ReadFile(filepath.Join(fixturesRoot, "mocks", "ai_service_"+size+".json"))
    modelRe := regexp.MustCompile(`"model":\s*"([^"]+)"`)
    rowRe := regexp.MustCompile(`"prompt":\s*"([^"]+)",\s*"output":\s*"([^"]+)",\s*"tokens":\s*(\d+)`)
    total := len(modelRe.FindStringSubmatch(string(data))[1])
    for _, match := range rowRe.FindAllStringSubmatch(string(data), -1) {
        tokens, _ := strconv.Atoi(match[3])
        total += len(match[1]) + len(match[2]) + tokens
    }
    fmt.Println(total)
}

File: tasks/implementations/php/tasks/ai_service_integration.php
<?php
function run_ai_service_integration(string $size, string $fixturesRoot): void {
  $text = file_get_contents($fixturesRoot . "/mocks/ai_service_{$size}.json");
  preg_match('/"model":\s*"([^"]+)"/', $text, $model);
  preg_match_all('/"prompt":\s*"([^"]+)",\s*"output":\s*"([^"]+)",\s*"tokens":\s*(\d+)/', $text, $matches, PREG_SET_ORDER);
  $total = strlen($model[1]);
  foreach ($matches as $match) $total += strlen($match[1]) + strlen($match[2]) + intval($match[3]);
  echo $total . PHP_EOL;
}
